src/api/spec/models/kiwi/image_spec.rb

```diff
@@ -76,4 +84,159 @@
       it { expect(subject.valid?).to be_falsey }
     end
   end
+
+  describe '.to_xml' do
+    context 'without a package' do
+      context 'without any repository' do
+        it { expect(kiwi_image.to_xml).to eq(Kiwi::Image::DEFAULT_KIWI_BODY) }
+      end
+
+      context 'with some repositories' do
+        before do
+          kiwi_image.repositories << create(:kiwi_repository)
+        end
+
+        subject { Nokogiri::XML::DocumentFragment.parse(kiwi_image.to_xml) }
+
+        it { expect(subject.errors).to be_empty }
+        it { expect(subject.xpath('.//image').length).to be(1) }
+        it { expect(subject.xpath('.//image/description').length).to be(1) }
+        it { expect(subject.xpath('.//image/packages').length).to be(1) }
+        it { expect(subject.xpath('.//image/repository').length).to be(1) }
+      end
+    end
+
+    context 'without kiwi image file' do
+      after do
+        # We need to be logged to destroy a package
+        login user
+        subject.package.destroy
```

Comment:
Why does the package need to be destroyed? Don't the rspec tests run in transactions which get rolled back once the test completes?